consensus/consensus-types/src/block.rs

```diff
@@ -156,7 +156,7 @@ where
             HashValue::zero(),
             ledger_info.transaction_accumulator_hash(),
             ledger_info.version(),
-            ledger_info.timestamp_usecs(),
+            0,
```

Comment:
How could ledger info timestamp be non-deterministic? Isn't it agreed upon all the participants? (I'm slightly uncomfortable with non-monotonic timestamps. We might have code ala `parent.timestamp + 1` or something like that for NIL blocks, need to update that).

Reply:
Because we could commit reconfiguraiton block directly with(t1) but also any descendent of it with t2, t3, t4, the timestamp of ledger info comes from the direct commit block.

re non-monotonic, it's still monotonic for every epoch, and also monotonic in storage ledger. The timestamp in genesis is just a placeholder that never gets persisted or exposed to anyone.

Reply:
Re non-determinism: right, I always forget about this one :).
Re timestamps: today the NIL blocks inherit the timestamps of their parents: can we accidentally have NIL block timestamp being `1` if there is a timeout in the first round? (do we have block metadata transactions for these blocks?)

Reply:
Maybe this is repetitive, but could we just use the timestamp of the last QC from the previous epoch? Or could we make this a constant GENESIS_TIMESTAMP?

Reply:
re NIL block - this problem exists for the first genesis too, but yeah that's not monotonic if it happens after and we do have block metadata for NIL.

we can not use the last QC's timestamp as there may be several different `last` QC, I'm fine with a GENESIS_TIMESTAMP but that doesn't solve the non-monotonic problem though.

One way to address this is let ledger info use the timestamp from executed state since we now persist the block metadata.

Reply:
Let's talk about it when you're in. I'd love to understand what executed state you're referring to (and yes, I'm all for using the execution timestamp corresponding to the actual reconfiguration for LI timestamp purposes).